Makefile

```diff
@@ -0,0 +1,111 @@
+# Run `make help` to display help
+.DEFAULT_GOAL := help
+
+# --- Global -------------------------------------------------------------------
+O = out
+COVERAGE = 0
+VERSION ?= $(shell git describe --tags --dirty --always)
+
+## Build and lint
+all: build lint
+	@if [ -e .git/rebase-merge ]; then git --no-pager log -1 --pretty='%h %s'; fi
+	@echo '$(COLOUR_GREEN)Success$(COLOUR_NORMAL)'
+
+## Full clean build and up-to-date checks as run on CI
+ci: clean check-uptodate all
+
+check-uptodate: tidy
+	test -z "$$(git status --porcelain -- go.mod go.sum)" || { git status; false; }
+
+## Remove generated files
+clean::
+	-rm -rf $(O) dist
```

Comment:
how come we use separate out and dist? wouldn't out/dist make sense?

Reply:
I dont really know much about `dist` - I think it is goreleaser that creates that. I just took this from the last project. Perhaps I can configure goreleaser to use `out/dist` - I'll have a look

Reply:
looks like you can:
https://goreleaser.com/customization/dist/
feel free to resolve conversation after reading.

Reply:
This is gone now. No more goreleaser. The directory used for creating release binaries is now `out/release`

Reply:
the camh way.